lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -23,6 +23,8 @@
 {-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE TypeOperators #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-} -- suppress false warning
```

Comment:
As a side note: I wish there were a way to suppress this on a function-by-function level. (Right now, we can only suppress this warning module-wide.)